examples/systemcalls/systemcalls.c

```diff
@@ -16,8 +23,16 @@ bool do_system(const char *cmd)
  *   and return a boolean true if the system() call completed with success 
  *   or false() if it returned a failure
 */
+    int wstatus;
 
-    return true;
+    /* Execute system command and check return value */
+    wstatus = system(cmd);
+    WIFEXITED(wstatus);
+
+    if (wstatus == true)
```

Comment:
@DhruvHMehta I'm surprised this works, I would have expected this to be 0 on success.

Reply:
Hi Dan, I read this in the manual
https://man7.org/linux/man-pages/man3/system.3.html

If all system calls succeed, then the return value is the
termination status of the child shell used to execute command.
(The termination status of a shell is the termination status
of the last command it executes.)

In the last two cases, the return value is a "wait status" that
can be examined using the macros described in waitpid(2).  (i.e.,
WIFEXITED(), WEXITSTATUS(), and so on).

Which is why I decided to use WIFEXITED to check the status.

Reply:
> Which is why I decided to use WIFEXITED to check the status.

But I think you'd need to do WIFEXECUTED(wstatus) to use this macro.

Reply:
Oh yes, I understand now. That is a mistake.